src/main.rs

```diff
@@ -44,22 +43,108 @@ async fn main() -> std::io::Result<()> {
         Arc::new(file_hosting::MockHost::new())
     };
 
-    // Get executable path
-    let mut exe_path = env::current_exe()?.parent().unwrap().to_path_buf();
-    // Create the path to the index lock file
-    exe_path.push("index.v1.lock");
-
-    // Indexing mods if not already done
-    if env::args().any(|x| x == "regen") {
-        // User forced regen of indexing
-        info!("Forced regeneration of indexes!");
-        index_mods(pool).await.expect("Mod indexing failed");
-    } else if !exe_path.exists() {
-        // The indexes were not created, or the version was upgraded
-        info!("Indexing of mods for first time...");
-        index_mods(pool).await.expect("Mod indexing failed");
-        // Create the lock file
-        File::create(exe_path)?;
+    // Currently removed to prevent double indexing:
+    // We may still want something similar, where it doesn't index
+    // on startup so that starting doesn't take a long time when
+    // it's being run frequently.
+
+    // // Get executable path
+    // let mut exe_path = env::current_exe()?.parent().unwrap().to_path_buf();
+    // // Create the path to the index lock file
+    // exe_path.push("index.v1.lock");
+
+    // let indexing_settings = IndexingSettings::from_env();
+
+    // // Indexing mods if not already done
+    // if env::args().any(|x| x == "regen") {
+    //     // User forced regen of indexing
+    //     info!("Forced regeneration of indexes!");
+    //     index_mods(pool.clone(), indexing_settings)
+    //         .await
+    //         .expect("Mod indexing failed");
+    // } else if !exe_path.exists() {
+    //     // The indexes were not created, or the version was upgraded
+    //     info!("Indexing of mods for first time...");
+    //     index_mods(pool.clone(), indexing_settings)
+    //         .await
+    //         .expect("Mod indexing failed");
+    //     // Create the lock file
+    //     File::create(exe_path)?;
+    // }
```

Comment:
Could we async index so the server still starts while the indexing is being done?

Reply:
That's what the system currently does.  The scheduler triggers the indexing once on its own threadpool/arbiter when the server is started, so the server starts while the indexing is being processed.  The main question with this part of the code is whether we want an option to not index at startup, so that rapid iteration and other development is easier without the indexing spamming the logs.
We could set up an environment variable to control whether the server indexes on startup and then every hour (the configured interval), or every hour after startup, but not at startup.

Reply:
I've used the argument system from before, but with `skip` instead of `regen` to allow the user to skip the initial mod indexing, which should cover what this code did before, besides checking the cache.

Reply:
I think it's fine the current way where it does it asynchronously. Is there a log output every time it indexes?